packages/core/src/report-diagnostic.ts

```diff
@@ -0,0 +1,37 @@
+import { StylableMeta } from './stylable-processor';
+
+export interface EmitDiagnosticsContext {
+    emitError(e: Error): void;
+    emitWarning(e: Error): void;
+}
+
+function reportDiagnostic(
```

Comment:
I think remapDiagnoticsByMode (or whatever name) would be a cleaner way of achieving this helper